Sources/TuistCore/Models/ArchiveAction.swift

```diff
@@ -0,0 +1,38 @@
+import Basic
+import Foundation
+
+public struct ArchiveAction: Equatable {
+    // MARK: - Attributes
+
+    public let configurationName: String
+    public let revealArchiveInOrganizer: Bool
+    public let customArchiveName: String?
+    public let preActions: [ExecutionAction]
+    public let postActions: [ExecutionAction]
+
+    // MARK: - Init
+
+    public init(
+        configurationName: String,
+        revealArchiveInOrganizer: Bool = true,
+        customArchiveName: String? = nil,
+        preActions: [ExecutionAction] = [],
+        postActions: [ExecutionAction] = []
+    ) {
+        self.configurationName = configurationName
+        self.revealArchiveInOrganizer = revealArchiveInOrganizer
+        self.customArchiveName = customArchiveName
+        self.preActions = preActions
+        self.postActions = postActions
+    }
+
+    // MARK: - Equatable
+
+    public static func == (lhs: ArchiveAction, rhs: ArchiveAction) -> Bool {
```

Comment:
All of `ArchiveAction` properties auto conform to equatable, the manual equatable will not be needed

Reply:
Right. I went through all the structs and deleted the manual conformance of the protocol. The only one that I couldn't remove was the `Target` model, because some attributes are tuples and they don't conform the equatable. In a follow-up PR I might turn those into structs instead.